### ISSUE TITLE
Value cannot be null

### ISSUE THREAD

**Issue by DieselTech** [yumiris/Gunloader#3]
Been following the project for some time now, nice to see your still keeping it updated!

Having an issue with the process. No matter what encoding format I pick, it seems to error out. This is the output from ogg but I've tried them all with the same problem. 

```
                            __                __
         ____ ___  ______  / /___  ____ _____/ /__  _____
        / __ `/ / / / __ \/ / __ \/ __ `/ __  / _ \/ ___/
       / /_/ / /_/ / / / / / /_/ / /_/ / /_/ /  __/ /
       \__, /\__,_/_/ /_/_/\____/\__,_/\__,_/\___/_/
      /____/
      ---------------------------------------------------
      author miris ~ github yumiris/youtube.album.extract
      ---------------------------------------------------
--------------------------------------------------------------------------------
Please go to https://gunloader.miris.design and follow the instructions there, then type the ID in here!
Gunloader ID: elephant-evil-7
--------------------------------------------------------------------------------
Please specify a format.
Available formats: flac | lame | opus | vorbis | aac | original
Format: opus
[youtube] _sI_Ps7JSEk: Downloading webpage
[youtube] _sI_Ps7JSEk: Downloading android player API JSON
[info] _sI_Ps7JSEk: Downloading 1 format(s): 248+251
[download] Destination: VIDEO~d7a4d27c-7920-4cab-97cc-6a750b78842d.f248.webm
[download] 100% of 194.47MiB in 00:10
[download] Destination: VIDEO~d7a4d27c-7920-4cab-97cc-6a750b78842d.f251.webm
[download] 100% of 50.45MiB in 00:02
[Merger] Merging formats into "VIDEO~d7a4d27c-7920-4cab-97cc-6a750b78842d.webm"
Deleting original file VIDEO~d7a4d27c-7920-4cab-97cc-6a750b78842d.f251.webm (pass -k to keep)
Deleting original file VIDEO~d7a4d27c-7920-4cab-97cc-6a750b78842d.f248.webm (pass -k to keep)
ffmpeg version 4.4-essentials_build-www.gyan.dev Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10.2.0 (Rev6, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libmfx --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
Input #0, matroska,webm, from 'VIDEO~d7a4d27c-7920-4cab-97cc-6a750b78842d.webm':
  Metadata:
    ENCODER         : Lavf58.76.100
  Duration: 00:49:49.62, start: -0.007000, bitrate: 687 kb/s
  Stream #0:0(eng): Video: vp9 (Profile 0), yuv420p(tv, bt709), 1920x1080, SAR 1:1 DAR 16:9, 25 fps, 25 tbr, 1k tbn, 1k tbc (default)
    Metadata:
      DURATION        : 00:49:49.600000000
  Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
    Metadata:
      DURATION        : 00:49:49.621000000
Stream mapping:
  Stream #0:0 -> #0:0 (vp9 (native) -> mjpeg (native))
Press [q] to stop, [?] for help
[swscaler @ 0000023bff7a9080] deprecated pixel format used, make sure you did set range correctly
Output #0, image2, to '1.jpg':
  Metadata:
    encoder         : Lavf58.76.100
  Stream #0:0(eng): Video: mjpeg, yuvj420p(pc, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], q=2-31, 200 kb/s, 25 fps, 25 tbn (default)
    Metadata:
      DURATION        : 00:49:49.600000000
      encoder         : Lavc58.134.100 mjpeg
    Side data:
      cpb: bitrate max/min/avg: 0/0/200000 buffer size: 0 vbv_delay: N/A
frame=    1 fps=0.0 q=11.6 Lsize=N/A time=00:00:00.04 bitrate=N/A speed=0.53x
video:200kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
ffmpeg version 4.4-essentials_build-www.gyan.dev Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10.2.0 (Rev6, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libmfx --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
Input #0, matroska,webm, from 'VIDEO~d7a4d27c-7920-4cab-97cc-6a750b78842d.webm':
  Metadata:
    ENCODER         : Lavf58.76.100
  Duration: 00:49:49.62, start: -0.007000, bitrate: 687 kb/s
  Stream #0:0(eng): Video: vp9 (Profile 0), yuv420p(tv, bt709), 1920x1080, SAR 1:1 DAR 16:9, 25 fps, 25 tbr, 1k tbn, 1k tbc (default)
    Metadata:
      DURATION        : 00:49:49.600000000
  Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
    Metadata:
      DURATION        : 00:49:49.621000000
Stream mapping:
  Stream #0:1 -> #0:0 (opus (native) -> pcm_s16le (native))
Press [q] to stop, [?] for help
Output #0, wav, to '1.wav':
  Metadata:
    ISFT            : Lavf58.76.100
  Stream #0:0(eng): Audio: pcm_s16le ([1][0][0][0] / 0x0001), 48000 Hz, stereo, s16, 1536 kb/s (default)
    Metadata:
      DURATION        : 00:49:49.621000000
      encoder         : Lavc58.134.100 pcm_s16le
size=   50620kB time=00:04:29.98 bitrate=1535.9kbits/s speed= 544x
video:0kB audio:50620kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.000150%
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'values')
   at System.String.JoinCore[T](ReadOnlySpan`1 separator, IEnumerable`1 values)
   at System.String.Join[T](Char separator, IEnumerable`1 values)
   at Gunloader.Encoders.Opus.Encode(Track track, FileInfo source, FileInfo cover)
   at Gunloader.Track.Encode(Toolkit toolkit, FileInfo video)
   at Gunloader.Albums.Album.Encode(Toolkit toolkit)
   at Gunloader.Program.Invoke()
   at Gunloader.Program.Wizard()
   at Gunloader.Program.Main(String[] args)
```

**Comment by yumiris** [comment]
Hey friend, cheers for the detailed report and words of encouragement!

I've released [v0.8.1]() just now, which hopefully fixes the issue you were facing. I suspect the problem was that one of the songs didn't have artists assigned to it.

If you're still encountering quirks, please don't hesitate to attach the generated `.gun` file here (it's a JSON file which you can open in a text editor to copy and paste).

**Comment by DieselTech** [comment]
> I've released [v0.8.1](https://github.com/yumiris/Gunloader/releases/tag/v0.8.1) just now, which hopefully fixes the issue you were facing. I suspect the problem was that one of the songs didn't have artists assigned to it.

That seems to to have fixed it! Thanks for the quick turnaround!
